### ISSUE TITLE
expo-updates crash on bare workflow

### ISSUE THREAD

**Issue by arqex** [expo/expo#8583]
### Environment

Expo 37, expo-updates 0.2.5. Android Q.

### Steps to Reproduce

I have a bare react native app, ejected from expo some time ago. I have added expo-updates following the instructions.

Everything works ok when I run `react-native run-android --variant=release`, the app works properly.

Then I do `expo publish` to create a new update.

When the app start, it just crashes and get closed. From `adb logcat`:

```
05-31 19:16:45.190 31901 31959 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
05-31 19:16:45.190 31901 31959 E AndroidRuntime: Process: com.discovmobile, PID: 31901
05-31 19:16:45.190 31901 31959 E AndroidRuntime: com.facebook.react.common.JavascriptException: Error: The Expo SDK requires Expo to run. It appears the native Expo modules are unavailable and this code is not running on Expo. Visit https://docs.expo.io to learn more about developing an Expo project., stack:
05-31 19:16:45.190 31901 31959 E AndroidRuntime: <unknown>@13:185
05-31 19:16:45.190 31901 31959 E AndroidRuntime: v@2:1474
05-31 19:16:45.190 31901 31959 E AndroidRuntime: <unknown>@9:67
05-31 19:16:45.190 31901 31959 E AndroidRuntime: v@2:1474
05-31 19:16:45.190 31901 31959 E AndroidRuntime: <unknown>@8:191
05-31 19:16:45.190 31901 31959 E AndroidRuntime: v@2:1474
05-31 19:16:45.190 31901 31959 E AndroidRuntime: <unknown>@6:48
05-31 19:16:45.190 31901 31959 E AndroidRuntime: v@2:1474
05-31 19:16:45.190 31901 31959 E AndroidRuntime: d@2:876
05-31 19:16:45.190 31901 31959 E AndroidRuntime: global code@1352:4
05-31 19:16:45.190 31901 31959 E AndroidRuntime:
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.modules.core.ExceptionsManagerModule.reportException(ExceptionsManagerModule.java:71)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:371)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:150)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:883)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:100)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:26)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:237)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:225)
05-31 19:16:45.190 31901 31959 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
```

### Expected Behavior

I expect the app to update its bundle without problems. I don't know why the app in its variant `release` needs the expo cli to work, it seems not to need it if I do `react-native run-android --variant=release`.

I'm using this other packages from expo:
```
    "expo": "^37.0.0",
    "expo-asset": "^8.1.5",
    "expo-constants": "~9.0.0",
    "expo-task-manager": "~8.1.0",
    "expo-updates": "^0.2.5",
```

### Actual Behavior

The app can't be used anymore. Need to uninstall and move it to the phone using `react-native` cli again.

### Reproducible Demo

https://github.com/arqex/otaTestApp

**Comment by brentvatne** [comment]
can you share your app privately with us? i'd be happy to investigate

**Comment by dineshmatta** [comment]
My Expo app is also crashing in the release mode on IOS devices. Its a brand new project started with Expo SDK 37 and then ejected in bare flow.

In debug mode app works fine in IOS simulator but crashes in release mode.

**Comment by brentvatne** [comment]
@dineshmatta - can you share your app privately with us? i'd be happy to investigate

**Comment by dineshmatta** [comment]
Hey @brentvatne I gave you access to my repo. Let me know if you need anything else.
FYI, running the app locally on simulator in release mode after expo Publish

**Comment by brentvatne** [comment]
i can look at it tomorrow! just heading to bed. in the meantime, you might want to give this article a read https://blog.expo.io/over-the-air-updates-from-expo-are-now-even-easier-to-use-376e2213fabf and make sure you’re on the latest version of expo updates, which is a bit easier to use than the initial release

**Comment by brentvatne** [comment]
@dineshmatta - you didn't configure your AppDelegate properly, please refer to https://github.com/expo/expo/blob/master/packages/expo-updates/README.md#appdelegateh in the expo-updates README for what you need to do

**Comment by dineshmatta** [comment]
Thanks @brentvatne for finding out the issue, after posting it yesterday night I too realized that my AppDelgate is not configured, I tried configuring it as per the docs, but was getting interfacing errors, I will try again today with latest version of expo-updates which you mentioned and then I will update you if I am still facing an issue.

**Comment by brentvatne** [comment]
the interface errors may be a result of the AppDelegate header file, be sure to configure that as well

**Comment by dineshmatta** [comment]
Hey @brentvatne 

I tried upgrading expo-updates to latest version and followed all the steps mentioned in the blog and documentation, now I am getting different error when building the project

Here are some of the error screenshots from my Xcode, also I have updated my branch so you can also check the code( you already have access to it)

![image](https://user-images.githubusercontent.com/1083007/83465700-c2ba2f80-a442-11ea-9a6b-5b78f30ba946.png)


![image](https://user-images.githubusercontent.com/1083007/83465715-cc439780-a442-11ea-840e-93a9a9a46a65.png)


![image](https://user-images.githubusercontent.com/1083007/83465720-d4033c00-a442-11ea-88e9-2337ab919549.png)


![image](https://user-images.githubusercontent.com/1083007/83465725-d9f91d00-a442-11ea-8397-935e8970d03c.png)


![image](https://user-images.githubusercontent.com/1083007/83465733-e2515800-a442-11ea-915f-e93bcc94b4aa.png)

**Comment by arqex** [comment]
Hey @brentvatne, thanks for having a look!

I'm trying to create a project to reproduce it. I can't share the main code, so I copied the main structure, removed the JS code and details:

https://github.com/arqex/otaTestApp

I can't reproduce the crash, because it seems I the updates are not working well in this project? I don't get a manifest from the Updates module and seems that it's never picking the updates. Even if they are there:

https://exp.host/@arqex/otatestapp/index.exp?sdkVersion=37.0.0

**Comment by dineshmatta** [comment]
Hi @brentvatne 

Did you get chance to look the error I am getting, still receiving the same error event without upgrading the expo-updates package

**Comment by brentvatne** [comment]
@dineshmatta - i would strongly advise initializing a new project with `npx create-react-native-app` and then comparing the AppDelegate.m and AppDelegate.h with your project to see where you made the mistake :)

**Comment by brentvatne** [comment]
@arqex - you don't get a manifest in development for bare projects or for your first build

see: https://blog.expo.io/over-the-air-updates-from-expo-are-now-even-easier-to-use-376e2213fabf - under the heading "Build-time updates have no manifest"

**Comment by arqex** [comment]
@brentvatne that explains why I was getting an empty manifest :)

I have updated the repo and now it reproduces issues error! I had to update all the package names from `com.whatever` to `com.otatestapi` to make it find the updates.

What I do to reproduce it:
* Install the app using `react-native run-android --variant=release`
* Update something in Root.tsx
* run `expo publish`
* reloading the app in the phone I can see how the update is being fetched and it ask me to restart
* I close the app and when I start it again it crashes
* I do a  `adb logcat *:E` and I can see in the logs:
```
06-03 10:47:12.855 27509 27552 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
06-03 10:47:12.855 27509 27552 E AndroidRuntime: Process: com.otatestapp, PID: 27509
06-03 10:47:12.855 27509 27552 E AndroidRuntime: com.facebook.react.common.JavascriptException: Error: The Expo SDK requires Expo to run. It appears the native Expo modules are unavailable and this code is not running on Expo. Visit https://docs.expo.io to learn more about developing an Expo project.
```

Do you need something to run an `expo publish` by yourself?

**Comment by brentvatne** [comment]
@arqex - do you have an ios and android directory in your project? do you have `react-native-unimodules` in your package.json? are you using the latest version of expo-cli?

**Comment by arqex** [comment]
Hey! Yes, ios and android dirs are in place, but I've only tried with android for this error. Here is the repo to reproduce it:

https://github.com/arqex/otaTestApp

`react-native-unimodules` is in package.json and I'm using `expo-cli` version 3.21.5.

**Comment by cem2ran** [comment]
@brentvatne Unfortunately we have the same issue. I'd be happy to provide access to our repo. I spent a few days trying to resolve this following all the instructions, but had to disable updates in the end. I'm confident that it has nothing to do with our application code as I also tried to replace our application entrypoint with a simple Text component wrapped in a SafeAreaView.

**Comment by byCedric** [comment]
Hey people! I can confirm this issue is related to `expo-cli@<=3.23.2`. This should be fixed in the latest CLI (`3.23.3`) 🚀   Please try that one and open a new issue when this happens again. Sorry for the inconvenience!
